File: kinotic-github/src/main/java/org/kinotic/github/api/model/GitHubInstallCompletion.java
package org.kinotic.github.api.model;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.Accessors;

/**
 * Result of finalising a GitHub install round-trip. Returned to the SPA's callback
 * component so it can drive the post-install UX. {@code returnTo} echoes what the
 * SPA staged when it called {@code startInstall(...)} and may carry query params
 * (e.g. {@code /projects?openNewProject=1}) so the destination page can pick up
 * where the user left off.
 */
@Getter
@Setter
@Accessors(chain = true)
@NoArgsConstructor
public class GitHubInstallCompletion {

    /** The persisted installation row. */
    private GitHubAppInstallation installation;

    /** SPA route the user wanted to land back on. */
    private String returnTo;
}

File: kinotic-js/workspace/packages/os-api/src/api/model/github/GitHubInstallCompletion.ts
import { GitHubAppInstallation } from '@/api/model/github/GitHubAppInstallation'

/**
 * Result of finalising a GitHub install round-trip. {@code returnTo} echoes what
 * the SPA staged when it called {@code startInstall(...)} and may carry query
 * params (e.g. {@code /projects?openNewProject=1}) so the destination page can
 * pick up where the user was.
 */
export class GitHubInstallCompletion {
    public installation!: GitHubAppInstallation
    public returnTo: string | null = null
}
